### PR TITLE
platform: fix unaligned 64-bit accesses on AArch32

### PR DESCRIPTION
**Pull request by stephenkyle-ARM**
Ensures that Aarch32 Arm builds with an Armv8 compiler do not set
BROTLI_64_BITS.

This scenario is possible with ChromeOS builds, as they may use a
toolchain with the target armv7-cros-gnueabi, but with -march=armv8.
This will set __ARM_ARCH to 8 (defining BROTLI_TARGET_ARMV8), but will
also set __ARM_32BIT_STATE and not __ARM_64BIT_STATE. Without this,
illegal 64-bit non-word-aligned reads (LDRD) may be emitted.

Also fix unaligned 64-bit reads on AArch32 - STRD was still possible to
emit.

### CODE REVIEW

**Comment by stephenkyle-ARM** [comment]
Relands google/brotli/699.

**Comment by stephenkyle-ARM** [comment]
The CI seems to suffer from network issues often... let me try again.

**Comment by stephenkyle-ARM** [comment]
Looks good now.

**Comment by eustas** [comment]
Thanks!
